clang/test/AST/ast-print-sycl-unique-stable-name.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clang_cc1 -ast-print -fsycl-is-device %s -o - -triple spir64-sycldevice | FileCheck %s
+// RUN: %clang_cc1 -ast-print -fsycl-is-device %s -o - -triple spir64 | FileCheck %s
```

Comment:
```suggestion
// RUN: %clang_cc1 -ast-print -fsycl-is-device %s -o - | FileCheck %s
```
@erichkeane, do we need to set the triple for AST test?

Reply:
> I probably just copy/pasted this triple.

Hm... this triple doesn't make sense to me. The second component of the triple is a "vendor" and there is no `sycldevice` vendor. It should be `spir64-unknown-unknown-sycldevice`. 

Anyway I created https://reviews.llvm.org/D108020 to clean up this test.

Reply:
My understanding is that if you 'skipped' `unknown` that the triple parser just figures it out, so making sure the arguments are correct position-ally isn't necessary.

Reply:
AFAIK, the 'skipped' version should still have `-`: `spir64---sycldevice`.

Reply:
Just by convention?  I've been using `spir64-sycldevice` for a while and it always seems to work, but I definitely never looked particularly close at it!  Anyway, I guess it is academic now :) I +1'ed your Phab review as well.